benchmarking/commons/hpo_main.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
There seems to be a lot of redundancy between the different hpo_*.py files. Would it make sense to merge them into a single file?

Reply:
I already have common functions in hpo_main.py. For the remainder, so hpo_main_*.py, they are really all a bit different in important ways. Any suggestion what could still be unified?

But they should rather be templates for people who want to modify them. You should select one of them, based on the back-end you are using. While we always say that this need 1 line of change only, this is not quite the full story, as you see from the differences in these 3 files.